assets/js/search.js

```diff
@@ -0,0 +1,124 @@
+(function() {
+    'use stict';
+    var activityCards, find, mapCardToClassName, mapCardToHeading, hideCard,
+        showCard, filterCards, onSeachChange, onClearSearch, createSearchInput,
+        activityContainer, searchInput;
+
+    activityCards = [].slice.call(document.querySelectorAll('.activity-card'));
+
+    find = function(needle, haystack) {
+        var regex;
+
+        regex = new RegExp(needle);
```

Comment:
I'd suggest to write this in one statement to make it easier to read. The same goes for the other functions as well where this pattern is used.